### PR TITLE
Refactored the class and fixed phpcs errors

### PR DESCRIPTION
**Pull request by dingo-d**
This PR is addressing the #89 issue.

I've refactored the class to its own file, made minor changes and pulled the file through the phpcs with `WordPress` coding standards (and ignored a few VIP ruleset issues).

### CODE REVIEW

**Comment by danieltj27** [comment]
Wow that's a pretty big diff. Thanks for doing this. I'll check it out soon when I have a chance 👍

**Comment by dingo-d** [comment]
I mostly removed the class, and made some minor changes (added `checked()` function for the checkbox, some strict checks etc.).

Happy to help :)

File: class-dark-mode.php [hunk 1]
```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * File holding the main class of the plugin.
+ *
+ * @since 2.1 Refactored the class to separate file.
```
Comment (danieltj27) [inline class-dark-mode.php]: Update to version 3.0.

File: class-dark-mode.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * File holding the main class of the plugin.
```
Comment (danieltj27) [inline class-dark-mode.php]: Change to:

```
The core Dark Mode class.
```

File: class-dark-mode.php [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+ */
+
+/**
+ * Main plugin class
+ */
+class Dark_Mode {
+	/**
+	 * Plugin version constant
+	 */
+	const PLUGIN_VERSION = '2.1';
```
Comment (danieltj27) [inline class-dark-mode.php]: Change to version 3.0.

File: class-dark-mode.php [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+	 * @since 1.6 Major rewrite to properly address automatic Dark Mode.
+	 * @since 2.0 Removed the automatic checks and added filters.
+	 *
+	 * @param int $user_id The user id to check.
+	 *
+	 * @return boolean
+	 */
+	public static function is_using_dark_mode( $user_id = 0 ) {
+		if ( 0 === $user_id ) {
+			// Default to the current user.
```
Comment (danieltj27) [inline class-dark-mode.php]: Remove this comment, we don't need it.

File: class-dark-mode.php [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+	/**
+	 * Add the stylesheet to the dashboard if enabled.
+	 *
+	 * @since 1.0
+	 * @since 2.1 Removed the register stylesheet function.
+	 *
+	 * @return void
+	 */
+	public static function load_dark_mode_css() {
+		// Get the current user id.
```
Comment (danieltj27) [inline class-dark-mode.php]: As above, remove this comment as it's self explanatory.

File: dark-mode.php [hunk 6]
```diff
@@ -8,283 +7,14 @@
  * Author URI: https://www.danieltj.co.uk/
  * Text Domain: dark-mode
  * Version: 2.1
+ *
+ * @package dark-mode
```
Comment (danieltj27) [inline dark-mode.php]: Let's not add this in the plugin header. We have it in the class file anyway.

File: dark-mode.php [hunk 7]
```diff
@@ -8,283 +7,14 @@
  * Author URI: https://www.danieltj.co.uk/
  * Text Domain: dark-mode
  * Version: 2.1
```
Comment (danieltj27) [inline dark-mode.php]: Bump to version 3.0.

File: dark-mode.php [hunk 8]
```diff
@@ -8,283 +7,14 @@
...
-			array_unshift( $links, $feedback_link );
-
-		}
-
-		return $links;
-
-	}
-
-}
+require 'class-dark-mode.php';
```
Comment (danieltj27) [inline dark-mode.php]: WordPress Core seems to use parenthesis a lot, if not exclusively so let's add them here.

```
require( 'class-dark-mode.php' );
```

File: dark-mode.php (hunk 6 above)
Comment (dingo-d) [reply dark-mode.php]: This was added to avoid the phpcs error `Missing @package tag in file comment` :)

File: dark-mode.php (hunk 8 above)
Comment (dingo-d) [reply dark-mode.php]: I'll add them. The WordPress coding standard seems to pull the `PEAR` standard for including files (without parentheses)

```sh
"require" is a statement not a function; no parentheses are required (PEAR.Files.IncludingFile.BracketsNotRequired)
```

**Comment by dingo-d** [comment]
I've made some changes, and added some comments. My wp.org profile is here: https://profiles.wordpress.org/dingo_bastard :)

File: class-dark-mode.php [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+ * @since 3.0 Refactored the class to separate file.
+ * @package dark-mode
+ */
+
+/**
+ * Main plugin class
+ */
+class Dark_Mode {
+	/**
+	 * Plugin version constant
```
Comment (danieltj27) [inline class-dark-mode.php]: Lets add a `@since` DocBlock syntax here for `3.0` and add full stop `.` to end of sentence.

Comment (dingo-d) [reply class-dark-mode.php]: I changed this now, is it ok?

**Comment by dingo-d** [comment]
Happy to help 😄

**Pull request merged**